File: 0001-two-sum/0001-two-sum.py
class Solution:
# O(n) time and O(n) space where n is the length of the nums list 
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        differences_dict = {} # {3: 0}

        for idx in range(len(nums)):
            curr_num = nums[idx] # 3
            difference = target - curr_num # 3
            if curr_num in differences_dict:
                return [differences_dict[curr_num], idx]
            differences_dict[difference] = idx

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 // O(n) time and O(n) space where n is the length of the input nums array
function twoSum(nums, target) {
    const numsSeen = {};
    for (let idx = 0; idx < nums.length; idx++) { 
      const currNum = nums[idx];
      const diff = target - currNum;
      if (diff in numsSeen) {
        return [numsSeen[diff], idx];
      }
      numsSeen[currNum] = idx;
    }
};